### DISCUSSION TITLE
While adding treatment, change "when" from numbers to a calendar pickdatetime to allow putting in future carbs

### DISCUSSION THREAD

**Discussion by KSayrs** [NightscoutFoundation/xDrip#3102] [Ideas]
My usual flow on Sugarmate was to add insulin when I took it, then add my carbs at the time I'd take them, usually around half an hour later. It's nice to do it then instead of later since the math is fresh in my mind.

This could also be taken into account for predictions

**Comment by Navid200** [comment]
What are you asking?  
Can you rewrite this thinking about the person who is reading it wanting to understand what you are suggesting not knowing everything in your mind?

**Comment by KSayrs** [comment]
When logging grams of carbs 
![UI-Treat2](https://github.com/NightscoutFoundation/xDrip/assets/14891390/00d726a8-1e26-4fc6-866e-7373203ee096)
And setting when 
![UI-Treat4](https://github.com/NightscoutFoundation/xDrip/assets/14891390/4a4081b3-861c-48f2-8320-fcfae79be6a0)

It only adds carbs to the past. 

I prebolus for around 30 minutes, so when I log my insulin, I would also like to log carbs, except that it will be another 30 minutes before I eat them.

I am asking for an option to add carbs for a future time.

Let me know if that is still not clear.

**Reply by jonprogers** [reply]
I think you already can - I've just tested a few and a sliding window is implemented to interpret the time entered in the 'when' section. It puts it in between -23hr to +1hr from the current time. I'm currently running the 30/07/2023 nightly.

**Reply by KSayrs** [reply]
Hmm, I might be on an older version then. I don't see a slider anywhere on the input screens

**Reply by jonprogers** [reply]
@KSayrs sorry, I obviously didn't explain well...
If it is say 18:00 currently and I add a treatment at 18:58 it enters the treatment 58 minutes in the future, although if I add a treatment at 19:02 it enters the treatment 22hrs 58mins in the past

**Reply by KSayrs** [reply]
Ahh I see -- so if it's within the same hour I can set it forward

**Reply by jonprogers** [reply]
That's what my test showed so, yes, I believe so. You can enter up to an hour in the future.

**Comment by Navid200** [comment]
@psonnera 
Is there no option to add carbs in the future?

**Reply by psonnera** [reply]
@Navid200 as answered above by @jonprogers . IMHO does more than 10/15 minutes in the future really make sense?

**Reply by KSayrs** [reply]
Within the hour is sufficient for my needs. Might be worth adding something to the docs about it though.

**Reply by psonnera** [reply]
@KSayrs will do. Thanks
